### PR TITLE
Abandon Session Mode «Load-Dependent Session Settings»

### PR DESCRIPTION
**Pull request by mBeym**
Removes everything related to the load-dependent session settings

Feature Wiki Page: https://docu.ilias.de/goto_docu_wiki_wpage_8263_1357.html

* @Uwe-Kohnle: We had to change one line in the SCORM components
* @kergomard: We removed the "Session Mode" setting from the user component

A quick review is highly appreciated.

### CODE REVIEW

File: components/ILIAS/Administration/classes/Setup/class.ilAdministrationDBUpdateSteps.php [hunk 1]
```diff
@@ -75,4 +75,42 @@ public function step_4(): void
     {
         $this->db->manipulate("DELETE FROM settings WHERE keyword LIKE 'char_selector%'");
     }
+
+    public function step_5(): void
```
Comment (mjansenDatabay) [inline components/ILIAS/Administration/classes/Setup/class.ilAdministrationDBUpdateSteps.php]: Please move this to an own database objective in the `Authentication` component. You can name it `AbandonLoadDependantSessionDatabaseUpdateObjective`.

File: components/ILIAS/Administration/classes/Setup/class.ilAdministrationDBUpdateSteps.php [hunk 2]
```diff
@@ -75,4 +75,42 @@ public function step_4(): void
...
+
+    public function step_6(): void
+    {
+
+    }
+
+    public function step_7(): void
+    {
+
+    }
```
Comment (mjansenDatabay) [inline components/ILIAS/Administration/classes/Setup/class.ilAdministrationDBUpdateSteps.php]: Why do we need these empty steps?

File: components/ILIAS/Authentication/classes/class.ilSession.php [hunk 3]
```diff
@@ -353,58 +334,28 @@ public static function _duplicate(string $a_session_id): string
      *
      * Returns the expiration timestamp in seconds
      *
-     * @param	boolean	If passed, the value for fixed session is returned
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilSession.php:382]: We should cleanup this comment in general.

File: components/ILIAS/Authentication/classes/class.ilSession.php [hunk 4]
```diff
@@ -353,58 +334,28 @@ public static function _duplicate(string $a_session_id): string
...
-        }
-        return time() + ilSessionControl::DEFAULT_MAX_IDLE * 60;
+            return time() + self::getIdleValue();
     }
 
     /**
      *
      * Returns the idle time in seconds
      *
-     * @param	boolean	If passed, the value for fixed session is returned
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilSession.php:414]: We should cleanup this comment in general.

File: components/ILIAS/Authentication/classes/class.ilSessionControl.php [hunk 5]
```diff
@@ -27,10 +27,7 @@ class ilSessionControl
      * default value for settings that have not
      * been defined in setup or administration yet
      */
-    public const DEFAULT_MAX_COUNT = 0;
     public const DEFAULT_MIN_IDLE = 15;
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilSessionControl.php:30]: Do we still need the "Min Idle Time" setting?

File: components/ILIAS/Authentication/templates/default/tpl.session_statistics_center.html [hunk 6]
```diff
@@ -12,6 +12,4 @@ <h3>{TIMEFRAME}</h3>
 			<li>{CAPTION_CLOSED_DETAILS}: {VALUE_CLOSED_DETAILS}</li>
 			<!-- END closed_details -->
 		</ul>		
-		<p>{CAPTION_MAXED_OUT_COUNTER}: {VALUE_MAXED_OUT_COUNTER}</p>
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/templates/default/tpl.session_statistics_center.html:15]: Maybe we should keep this, so the "Session Statistics" can work with the old values.

File: components/ILIAS/Authentication/classes/class.ilSession.php [hunk 7]
```diff
@@ -52,7 +34,6 @@ class ilSession
     public const SESSION_CLOSE_EXPIRE = 2;  // has expired
     public const SESSION_CLOSE_FIRST = 3;  // kicked by session control (first abidencer)
     public const SESSION_CLOSE_IDLE = 4;  // kickey by session control (ilde time)
-    public const SESSION_CLOSE_LIMIT = 5;  // kicked by session control (limit reached)
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/class.ilSession.php:55]: Why do we delete this but keep `SESSION_CLOSE_FIRST`?

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Thx @mBeym,

- [x] Please revert all language file related changes except those in `ilias_de.lang` and `ilias_en.lang`.
- [x] Please move the database update steps to the `Authentication` component.
- [x] In my opinion, we could remove more constants in `ilSession` but maybe have to migrate or even delete data/drop table columns from the session statistics database tables. I suggest activating the feature in “Fixed session duration” mode and checking which database fields can be dispensed with.

Best regards,
Michael

File: components/ILIAS/Authentication/classes/class.ilSessionControl.php (hunk 5 above)
Comment (mBeym) [reply components/ILIAS/Authentication/classes/class.ilSessionControl.php:30]: it's still being used in ilSCORM13PlayerGUI.

I could move that constant into the scorm class directly or replace it with hard coded values though.

File: components/ILIAS/Authentication/templates/default/tpl.session_statistics_center.html (hunk 6 above)
Comment (mBeym) [reply components/ILIAS/Authentication/templates/default/tpl.session_statistics_center.html:15]: I couldn't find any code line where this variable is still set.

File: components/ILIAS/Authentication/classes/class.ilSession.php (hunk 4 above)
Comment (mBeym) [reply components/ILIAS/Authentication/classes/class.ilSession.php:414]: just remove the `@return` or the entire comment?

**Comment by mjansenDatabay** [comment]
@mBeym Please resolve the conflicts

**Comment by mBeym** [comment]
> @mBeym Please resolve the conflicts

done

**Review by kergomard — APPROVED** [review]
Thank you very much for the PR @mBeym and @mjansenDatabay !

I'm fine with the changes in `ilObjUserFolderGUI`.

If you need to touch this once more (and only then, as it is not important enough): Might I ask you to rename the `$cb` variable [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-27c73a5b3b4e3d214fc68f669f302bbe963936859467d24c15a9eafab8f59606L2064). I know, it was called $cb, but I don't care much about it being a  checkbox, I would like to know what it actually sets ;-). Maybe something like ($session_reminder_cb or so).

And a very small suggestion:
Shouldn't we maybe drop the old settings through a db-update-step? I know, I'm a stickler for detail, but if for any reason you need to go through the settings table all old clutter just, well, clutters things up.

Thanks again and best,
@kergomard

**Comment by mjansenDatabay** [comment]
> Thank you very much for the PR @mBeym and @mjansenDatabay !
> 
> I'm fine with the changes in `ilObjUserFolderGUI`.
> 
> If you need to touch this once more (and only then, as it is not important enough): Might I ask you to rename the `$cb` variable [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-27c73a5b3b4e3d214fc68f669f302bbe963936859467d24c15a9eafab8f59606L2064). I know, it was called $cb, but I don't care much about it being a checkbox, I would like to know what it actually sets ;-). Maybe something like ($session_reminder_cb or so).
> 
> And a very small suggestion: Shouldn't we maybe drop the old settings through a db-update-step? I know, I'm a stickler for detail, but if for any reason you need to go through the settings table all old clutter just, well, clutters things up.
> 
> Thanks again and best, @kergomard

Thx @kergomard , the old settings should be deleted here: https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-8b1ca2c61f7661b03b2012f2337b9baba324b4eb4ebb25c5128b3ea40f55d72aR37

@mBeym Can you please rename the variable?

**Comment by kergomard** [comment]
Thank you very much for the pointer @mjansenDatabay ! Sorry for not having read carefully enough (one should not collapse the files overview and forget one has done so)!

**Comment by mBeym** [comment]
> > Thank you very much for the PR @mBeym and @mjansenDatabay !
> > I'm fine with the changes in `ilObjUserFolderGUI`.
> > If you need to touch this once more (and only then, as it is not important enough): Might I ask you to rename the `$cb` variable [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-27c73a5b3b4e3d214fc68f669f302bbe963936859467d24c15a9eafab8f59606L2064). I know, it was called $cb, but I don't care much about it being a checkbox, I would like to know what it actually sets ;-). Maybe something like ($session_reminder_cb or so).
> > And a very small suggestion: Shouldn't we maybe drop the old settings through a db-update-step? I know, I'm a stickler for detail, but if for any reason you need to go through the settings table all old clutter just, well, clutters things up.
> > Thanks again and best, @kergomard
> 
> Thx @kergomard , the old settings should be deleted here: https://github.com/ILIAS-eLearning/ILIAS/pull/7525/files#diff-8b1ca2c61f7661b03b2012f2337b9baba324b4eb4ebb25c5128b3ea40f55d72aR37
> 
> @mBeym Can you please rename the variable?

Done. Renamed to **$session_reminder**.

@kergomard I also renamed the **$ti** ilNonEditableValueGUI variable to **$session_config** if that's ok for you?

**Review by mjansenDatabay — APPROVED** [review]

**Comment by mjansenDatabay** [comment]
Hi @pascalseeland ,

I would like to merge this within the next week. Any feedback is highly appreciated.

Best regards,
Michael

**Review by pascalseeland — APPROVED** [review]

**Pull request merged**